LandMark/Feature/Home/Pin/Data/Location/LocationManager.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  LocationManager.swift
+//  LandMark
+//
+//  Created by Lucas Migge on 31/10/24.
+//
+
+import CoreLocation
+
+protocol LocationManagerProtocol {
+    func getCurrentUserLocation() -> LocationCoordinate
+}
+
+class LocationManager: NSObject, CLLocationManagerDelegate, ObservableObject, LocationManagerProtocol {
+    private let locationManager = CLLocationManager()
+    
+    @Published var userCoordinate: LocationCoordinate = .init(latitude: 0, longitude: 0)
```

Comment:
Baseado no comentário acima sobre o nome dessa struct:
```suggestion
    @Published var userCoordinates: LocationCoordinates = .init(latitude: 0, longitude: 0)
```